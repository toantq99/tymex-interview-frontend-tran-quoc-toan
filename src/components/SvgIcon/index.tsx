import { FC, HTMLAttributes } from 'react'

const SvgIcon: FC<HTMLAttributes<HTMLImageElement> & { iconName: string }> = ({
  iconName,
  ...props
}) => <img src={`/assets/icons/${iconName}.svg`} {...props} />

export default SvgIcon
