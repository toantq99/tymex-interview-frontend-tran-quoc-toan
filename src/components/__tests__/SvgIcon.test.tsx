import { render, screen } from '@testing-library/react'
import SvgIcon from '../SvgIcon'

describe('SvgIcon Component', () => {
  it('renders an image with the correct src based on iconName', () => {
    const iconName = 'online'
    render(<SvgIcon iconName={iconName} />)

    const imgElement = screen.getByRole('img', { name: iconName })
    expect(imgElement).toBeInTheDocument()
    expect(imgElement).toHaveAttribute('src', `/assets/icons/${iconName}.svg`)
  })

  it('renders an image with the correct alt attribute', () => {
    const iconName = 'offline'
    render(<SvgIcon iconName={iconName} />)

    const imgElement = screen.getByRole('img', { name: iconName })
    expect(imgElement).toHaveAttribute('alt', iconName)
  })

  it('passes additional props to the img element', () => {
    const iconName = 'status'
    const title = 'User Status Icon'
    render(<SvgIcon iconName={iconName} title={title} />)

    const imgElement = screen.getByRole('img', { name: iconName })
    expect(imgElement).toHaveAttribute('title', title)
  })

  it('renders match snap shot', () => {
    const iconName = 'ethereum'

    render(<SvgIcon iconName={iconName} />)

    const imgElement = screen.getByRole('img', { name: iconName })
    expect(imgElement).toMatchSnapshot()
  })
})
