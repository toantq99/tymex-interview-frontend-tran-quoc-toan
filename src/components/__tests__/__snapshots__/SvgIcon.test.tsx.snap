// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon Component renders match snap shot 1`] = `
<img
  alt="ethereum"
  src="/assets/icons/ethereum.svg"
/>
`;
